numpy/linalg/tests/test_linalg.py

```diff
@@ -1761,6 +1763,40 @@ def test_xerbla_override():
             raise SkipTest('Numpy xerbla not linked in.')
 
 
+def test_sdot_bug_8577():
+    # Regression test that loading certain other libraries does not
+    # result to wrong results in float32 linear algebra.
+    #
+    # There's a bug gh-8577 on OSX that can trigger this, and perhaps
+    # there are also other situations in which it occurs.
+    #
+    # Do the check in a separate process.
+
+    bad_libs = ['PyQt5.QtWidgets', 'IPython']
+
+    template = textwrap.dedent("""
+    import sys
+    {before}
+    try:
+        import {bad_lib}
+    except ImportError:
+        sys.exit(0)
+    {after}
+    x = np.ones(2, dtype=np.float32)
+    sys.exit(0 if np.allclose(x.dot(x), 2.0) else 1)
```

Comment:
Don't you need the sqrt here to trigger the error, based on the comments in the issue?

Reply:
At least for #8577 sqrt wasn't mentioned

Reply:
Also #11166 reports it fails for just `dot`, so I think the test as it is now catches those reported cases.

Reply:
Thanks for checking